src/Compilers/CSharp/Test/Semantic/Semantics/ColorColorTests.cs

```diff
@@ -1729,6 +1730,37 @@ class X
             CompileAndVerify(comp, expectedOutput: "42");
         }
 
+        [WorkItem(5362, "https://github.com/dotnet/roslyn/issues/5362")]
+        [Fact]
+        public void TestColorColorSymbolInfoInArrowExpressionClauseSyntax()
+        {
+            const string source = @"public enum Lifetime
+{
+    Persistent,
+    Transient,
+    Scoped
+}
+public class Example
+{
+    public Lifetime Lifetime => Lifetime.Persistent;
+    //                          ^^^^^^^^
+}";
+            var comp = CreateCompilationWithMscorlibAndSystemCore(source, options: TestOptions.ReleaseDll);
+            comp.VerifyDiagnostics();
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var syntaxRoot = syntaxTree.GetRoot();
+
+            var semanticModel = comp.GetSemanticModel(syntaxTree, false);
+
+            var nameSyntax = syntaxRoot.FindNode(TextSpan.FromBounds(130, 138));
+            Assert.Equal("Lifetime", nameSyntax.ToString());
+
+            var actualSymbol = semanticModel.GetSymbolInfo(nameSyntax);
+
+            Assert.Equal("Lifetime", actualSymbol.Symbol.ToTestDisplayString());
```

Comment:
Since you decided to remove GetDeclaredSymbol part, consider also adding Assert.Equal for actualSymbol.Symbol.Kind to make it more clear that it is a type.